File: backend/src/main/java/com/campuslostfound/model/Item.java
package com.campuslostfound.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Entity
@Table(name = "items")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @NotBlank(message = "Name is required")
    @Size(min = 1, max = 100, message = "Name must be between 1 and 100 characters")
    private String name;

    @Column(columnDefinition = "TEXT")
    @Size(max = 1000, message = "Description must not exceed 1000 characters")
    private String description;

    @Column(nullable = true)
    @Size(max = 50, message = "Category must not exceed 50 characters")
    private String category;

    @Column(name = "image_url")
    private String imageUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @NotNull(message = "Status is required")
    private ItemStatus status = ItemStatus.LOST;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "report_id", nullable = false, foreignKey = @ForeignKey(name = "fk_item_report"))
    @NotNull(message = "Report is required")
    private Report report;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "claimed_by_user_id", foreignKey = @ForeignKey(name = "fk_item_claimed_user"))
    private User claimedByUser;

    @Column(name = "created_at", nullable = false, updatable = false, insertable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private Timestamp createdAt;
}

File: frontend/src/models/item.ts
import type { Report, ReportStatus, ReportType } from "./report";
import type { User } from "./user";

export interface Item {
    id: number;
    name: string;
    description?: string;
    reportId: number;
    claimedByUserId?: number;
    createdAt: string;
    status: ItemStatus;
    locationId?: number;
    locationName?: string;
    reportStatus?: ReportStatus;
    reportType?: ReportType
    reporterUserId?: number;
    reporterUserName?: string;
    report?: Report;
    claimedByUser?: User;
}

export interface ItemCreateData {
    name: string;
    description?: string;
    reportId: number;
}

export interface ItemUpdateData {
    name?: string;
    description?: string;
}

export interface ItemFilters {
    reportId?: number;
    claimedByUserId?: number;
}

export enum ItemStatus {
    CLAIMED = "CLAIMED",
    UNCLAIMED = "UNCLAIMED"
}
